compiler/src/iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -463,6 +465,33 @@ static LogicalResult adaptComputeConsumerToAvoidStackAllocation(
   return success();
 }
 
+/// Replaces a tensor.empty op with bufferization.alloc_tensor op which is
+/// created by tiling tensor.unpack op. It is intended because tiling unpack ops
+/// with non-perfect sizes needs extra elements. See the tiling implementation
+/// of tensor.unpack op for more details.
+static LogicalResult replaceUnpackEmptyWithAllocTensor(OpBuilder &b,
+                                                       func::FuncOp funcOp) {
+  funcOp.walk<WalkOrder::PreOrder>([&](tensor::EmptyOp emptyOp) {
+    bool isUsedByNonPerfectUnpack = false;
```

Comment:
Wouldn't this be easier if you just walk for the unpack operation and look at operand that is a `tensor.empty`. (You can delete the `tensor.empty` if it has no other uses).